File: old_trunk/cost/iona/orbacus/java/main/TestIntfExceptionsExt_2_3_impl.java
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

package ORBTest;

import org.omg.CORBA.*;
import org.omg.PortableServer.*;

final class TestIntfExceptionsExt_2_3_impl
    extends ORBTest_ExceptionsExt_2_3.IntfPOA
{
    private POA m_poa;

    public
    TestIntfExceptionsExt_2_3_impl(
	POA poa
    )
    {
	m_poa = poa;
    }

    public synchronized void
    op_CODESET_INCOMPATIBLE_Ex()
    {
	throw new CODESET_INCOMPATIBLE(31, CompletionStatus.COMPLETED_NO);
    }

    public synchronized void
    op_REBIND_Ex()
    {
	throw new REBIND(32, CompletionStatus.COMPLETED_NO);
    }

    public synchronized void
    op_TIMEOUT_Ex()
    {
	throw new TIMEOUT(33, CompletionStatus.COMPLETED_NO);
    }

    public synchronized void
    op_TRANSACTION_UNAVAILABLE_Ex()
    {
	throw new TRANSACTION_UNAVAILABLE(34, CompletionStatus.COMPLETED_NO);
    }

    public synchronized void
    op_TRANSACTION_MODE_Ex()
    {
	throw new TRANSACTION_MODE(35, CompletionStatus.COMPLETED_NO);
    }

    public synchronized void
    op_BAD_QOS_Ex()
    {
	throw new BAD_QOS(36, CompletionStatus.COMPLETED_NO);
    }

    public org.omg.PortableServer.POA
    _default_POA()
    {
	return m_poa;
    }
}

File: old_trunk/cost/iona/orbacus/cpp/main/TestIntfExceptionsExt_2_3_impl.cpp
// **********************************************************************
//
// Copyright (c) 2001
// IONA Technologies, Inc.
// Waltham, MA, USA
//
// All Rights Reserved
//
// **********************************************************************

#include <includes.h>

#include <TestIntfExceptionsExt_2_3_impl.h>

#include <limits.h>

using namespace CORBA;

TestIntfExceptionsExt_2_3_impl::TestIntfExceptionsExt_2_3_impl()
{
}

void
TestIntfExceptionsExt_2_3_impl::op_CODESET_INCOMPATIBLE_Ex()
    throw (CORBA::SystemException)
{
    throw CODESET_INCOMPATIBLE(31, COMPLETED_NO);
}

#ifndef HAVE_NO_CORBA_2_4

void
TestIntfExceptionsExt_2_3_impl::op_REBIND_Ex()
    throw (CORBA::SystemException)
{
    throw REBIND(32, COMPLETED_NO);
}

void
TestIntfExceptionsExt_2_3_impl::op_TIMEOUT_Ex()
    throw (CORBA::SystemException)
{
    throw TIMEOUT(33, COMPLETED_NO);
}

void
TestIntfExceptionsExt_2_3_impl::op_TRANSACTION_UNAVAILABLE_Ex()
    throw (CORBA::SystemException)
{
    throw TRANSACTION_UNAVAILABLE(34, COMPLETED_NO);
}

void
TestIntfExceptionsExt_2_3_impl::op_TRANSACTION_MODE_Ex()
    throw (CORBA::SystemException)
{
    throw TRANSACTION_MODE(35, COMPLETED_NO);
}

void
TestIntfExceptionsExt_2_3_impl::op_BAD_QOS_Ex()
    throw (CORBA::SystemException)
{
    throw BAD_QOS(36, COMPLETED_NO);
}

#endif
